File: fix_extra_braces.py
#!/usr/bin/env python3
import os
import re
import glob

def fix_extra_braces(file_path):
    """Fix extra closing braces in TypeScript/JavaScript files"""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        original_content = content
        
        # Fix pattern: </> followed by ); followed by }; followed by };
        # This should be: </> followed by ); followed by };
        pattern = r'(\s*</>\s*\);\s*\};\s*\};\s*)(\n\s*export default)'
        replacement = r'  );\n};\n\2'
        content = re.sub(pattern, replacement, content)
        
        # Fix pattern: </> followed by ); followed by }; followed by }; (without newline)
        pattern = r'(\s*</>\s*\);\s*\};\s*\};\s*)(export default)'
        replacement = r'  );\n};\n\2'
        content = re.sub(pattern, replacement, content)
        
        # Only write if content changed
        if content != original_content:
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(content)
            return True
            
        return False
    except Exception as e:
        print(f"Error fixing {file_path}: {e}")
        return False

def main():
    # Find all TypeScript/JavaScript files in the app directory
    patterns = [
        'app/**/*.tsx',
        'app/**/*.ts',
        'app/**/*.js',
        'app/**/*.jsx'
    ]
    
    files_fixed = 0
    total_files = 0
    
    for pattern in patterns:
        for file_path in glob.glob(pattern, recursive=True):
            total_files += 1
            if fix_extra_braces(file_path):
                files_fixed += 1
                print(f"Fixed extra braces in: {file_path}")
    
    print(f"Fixed extra braces in {files_fixed} out of {total_files} files")

if __name__ == "__main__":
    main()

File: fix-extra-braces.js
#!/usr/bin/env node

import fs from 'fs';
import path from 'path';

console.log('🔧 Fixing extra braces...');

// Function to fix extra braces
function fixExtraBraces(content) {
  // Remove extra closing braces at end of lines
  content = content.replace(/;}\n/g, ';\n');
  content = content.replace(/}\n/g, '\n');
  
  // Fix specific patterns
  content = content.replace(/'use client';}\n/g, "'use client';\n");
  content = content.replace(/import.*?;}\n/g, (match) => match.replace(';}', ';'));
  
  return content;
}

// Function to process a single file
function processFile(filePath) {
  try {
    let content = fs.readFileSync(filePath, 'utf8');
    const originalContent = content;
    
    // Apply fixes
    content = fixExtraBraces(content);
    
    // Only write if content changed
    if (content !== originalContent) {
      fs.writeFileSync(filePath, content, 'utf8');
      console.log(`✅ Fixed: ${filePath}`);
      return true;
    }
    
    return false;
  } catch (error) {
    console.error(`❌ Error processing ${filePath}:`, error.message);
    return false;
  }
}

// Function to find all TypeScript/JSX files
function findFiles(dir, extensions = ['.tsx', '.ts', '.jsx', '.js']) {
  const files = [];
  
  function traverse(currentDir) {
    const items = fs.readdirSync(currentDir);
    
    for (const item of items) {
      const fullPath = path.join(currentDir, item);
      const stat = fs.statSync(fullPath);
      
      if (stat.isDirectory()) {
        // Skip certain directories
        if (!['node_modules', '.git', '.next', 'dist', 'build'].includes(item)) {
          traverse(fullPath);
        }
      } else if (extensions.some(ext => item.endsWith(ext))) {
        files.push(fullPath);
      }
    }
  }
  
  traverse(dir);
  return files;
}

// Main execution
async function main() {
  try {
    console.log('📁 Finding files to process...');
    const files = findFiles('./app');
    
    console.log(`📊 Found ${files.length} files to process`);
    
    let fixedCount = 0;
    
    for (const file of files) {
      try {
        if (processFile(file)) {
          fixedCount++;
        }
      } catch (error) {
        console.error(`Error processing ${file}:`, error.message);
      }
    }
    
    console.log(`\n📈 Results:`);
    console.log(`✅ Fixed: ${fixedCount} files`);
    
    console.log('\n🎉 Extra braces fix completed!');
    
  } catch (error) {
    console.error('💥 Fatal error:', error.message);
    process.exit(1);
  }
}

main();